src/ssl.c

```diff
@@ -5584,6 +5585,18 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
     if (format == WOLFSSL_FILETYPE_PEM) {
     #ifdef WOLFSSL_PEM_TO_DER
         ret = PemToDer(buff, sz, type, &der, heap, info, &keyFormat);
+        #ifdef HAVE_PKCS8
```

Comment:
This block is repeated in the `else` of this `if (format == WOLFSSL_FILETYPE_PEM)` conditional. I think you can move this block out of the conditional, since it should happen either way:
```
diff --git a/src/ssl.c b/src/ssl.c
index 645d21cd2..ccfe7eb87 100644
--- a/src/ssl.c
+++ b/src/ssl.c
@@ -5544,6 +5544,7 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
 #if (defined(WOLFSSL_ENCRYPTED_KEYS) && !defined(NO_PWDBASED)) || \
      defined(HAVE_PKCS8)
     word32        algId = 0;
+    word32        crvId = 0;
 #endif
 #ifdef WOLFSSL_SMALL_STACK
     EncryptedInfo* info = NULL;
@@ -5614,20 +5615,21 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
             if (ret == 0) {
                 XMEMCPY(der->buffer, buff, length);
             }
+        }
+    }
 
-        #ifdef HAVE_PKCS8
-            /* if private key try and remove PKCS8 header */
-            if (type == PRIVATEKEY_TYPE) {
-                if ((ret = ToTraditional_ex(der->buffer, der->length, &algId)) > 0) {
-                    /* Found PKCS8 header */
-                    /* ToTraditional_ex moves buff and returns adjusted length */
-                    der->length = ret;
-                }
-                ret = 0; /* failures should be ignored */
-            }
-        #endif
+#ifdef HAVE_PKCS8
+    /* if private key try and remove PKCS8 header */
+    if (type == PRIVATEKEY_TYPE) {
+        if ((ret = ToTraditional_ex(der->buffer, der->length,
+                                                 &algId, &crvId, 1)) > 0) {
+            /* Found PKCS8 header */
+            /* ToTraditional_ex moves buff and returns adjusted length */
+            der->length = ret;
         }
+        ret = 0; /* failures should be ignored */
     }
+#endif
 
     if (used) {
         *used = info->consumed;
```

Reply:
Yes, you are correct. The block can move under if-else. Thanks